Jint/Engine.cs

```diff
@@ -162,6 +166,54 @@ public Engine(Action<Options> options)
             ClrTypeConverter = new DefaultTypeConverter(this);
             BreakPoints = new List<BreakPoint>();
             DebugHandler = new DebugHandler(this);
+
+            BuildExtensionMethodTypeCache();
+            RegisterCachedExtensionMethods();
+        }
+
+        internal Dictionary<Type, List<MethodInfo>> ExtensionMethodTypeCache = new Dictionary<Type, List<MethodInfo>>();
+
+        private void BuildExtensionMethodTypeCache()
+        {
+            if (Options._ExtensionMethodsTypes.Any())
+            {
+                foreach (var extensionMethodType in Options._ExtensionMethodsTypes)
+                {
+                    List<MethodInfo> extMethodType;
+                    if (!ExtensionMethodTypeCache.TryGetValue(extensionMethodType, out extMethodType))
+                    {
+                        foreach (var methodInfo in extensionMethodType.GetExtensionMethods())
+                        {
+                            var firstParamType = methodInfo.GetParameters().First().ParameterType;
+
+                            if (ExtensionMethodTypeCache.ContainsKey(firstParamType))
+                            {
+                                ExtensionMethodTypeCache[firstParamType].Add(methodInfo);
+                            }
+                            else
+                            {
+                                ExtensionMethodTypeCache.Add(firstParamType, new List<MethodInfo>() { methodInfo });
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        private void RegisterCachedExtensionMethods()
+        {
+            if (ExtensionMethodTypeCache.Any())
+            {
+                foreach (var cachedType in ExtensionMethodTypeCache)
+                {
+                    if (cachedType.Key == typeof(String))
+                    {
+                        foreach (var methodInfo in cachedType.Value)
+                        {
+                            String.PrototypeObject.FastAddProperty(methodInfo.Name, new MethodInfoFunctionInstance(this, new MethodInfo[] { methodInfo }), true, false, true);
```

Comment:
Probably not. Don't change the string prototype object. This is different from just supporting extension methods.